From 638f7a69c73799f2e904aff18908cadad1556a6e Mon Sep 17 00:00:00 2001
From: Christoph Hellwig <hch@lst.de>
Date: Tue, 9 Jan 2018 15:00:35 +0100
Subject: [PATCH 19/49] riscv: enable HAVE_FUTEX_CMPXCHG if !SMP

We use the asm-generic implementation, so it will always be there.

XXX: this works around the fact that the probe trips up an unresolved
pagefaul with split supervisor/user address spaces.

Signed-off-by: Christoph Hellwig <hch@lst.de>
Signed-off-by: Palmer Dabbelt <palmer@sifive.com>
---
 arch/riscv/Kconfig | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/arch/riscv/Kconfig b/arch/riscv/Kconfig
index 1afd0fb..b70918e 100644
--- a/arch/riscv/Kconfig
+++ b/arch/riscv/Kconfig
@@ -25,6 +25,8 @@ config RISCV
 	select HAVE_MEMBLOCK_NODE_MAP
 	select HAVE_DMA_API_DEBUG
 	select HAVE_DMA_CONTIGUOUS
+	# XXX: needs a proper implementation for SMP
+	select HAVE_FUTEX_CMPXCHG if !SMP
 	select HAVE_GENERIC_DMA_COHERENT
 	select IRQ_DOMAIN
 	select NO_BOOTMEM
-- 
2.7.4

